feat: draft code for auto and change name to BasketAuto

gwarby opened the pull request:
wrote initial draft for autonomous code dropping off preloaded sample and picking up two more to score, then parking in submersible

gwarby committed:
feat: draft code for auto and change name to general autonomous

wrote initial draft for autonomous code dropping off preloaded sample and picking up two more to score, then parking in submersible

gwarby committed:
feat: new name BasketAuto fix: syntax errors

still saying "ftc.pose" on null object reference on line 50 when  trying to get x value. Maybe create new functions to get specific x, y and yaw values?

gwarby commented:
@lcdnbl if you get some free time, can you look through this code and let me know if you can figure out why it would be giving me an error (null object reference) on line 50? if not, all good.

lcdnbl commented:
> @lcdnbl if you get some free time, can you look through this code and let me know if you can figure out why it would be giving me an error (null object reference) on line 50? if not, all good.

@gwarby this code is a little dangerous, because getCurrentAprilTagIds() checks for getDetections(), then getAprilTagDetection(int) does getDetections() again.  It could be null the 2nd time but not the 1st, causing line 50 to error when it tries to .ftcPose on a null.

Looking through the code:
  - on line 50, lib shouldn't be null because line 48 used lib & line 23 created it
  - on line 50, I don't know whether an ftcPose (which is non-null) can ever lack a y, but I wouldn't put effort into investigating that
  - so that brings us back to, most likely, lib.getAprilTagDetection(mainTagId) gave you back null - which it will do if the list of detections it asks for doesn't have mainTagId in the .id values
  - there is a tiny mismatch between the List<Integer> that came back on line 46 & the int you .get from it on line 48
  - I would definitely protect line 50 by inserting something like:
```
double distanceFromWall = 0.0;
AprilTagDetection tag = lib.getAprilTagDetection(mainTagId)
if (tag == null) {
  telemetry.addLine("April tag unexpectedly disappeared!");
} else {
   distanceFromWall = tag.ftcPose.y;
   telemetry.addData("Distance from wall: ", distanceFromWall);
}
```
One of the main goals of this, is to let the code run long enough so you get to see what was printed by line 49.  
It would be easier to just put `telemetry.update();` and `sleep(5000);` in before line 50.

gwarby committed:
feat: add catch if we don't have any info from April tag

gwarby committed:
feat: check for April tags 100 times

gwarby committed:
fix: got April tags working

What we needed to do was wait for everything to initialize and get going, so we have an if statement that checks if the output is null, and if it is, doesn't request anything more, but if the output is not null, then it reads the y value (distance from wall), and the last known distance is the one saved and used when running the program. New problem is that motors will not run quickly enough, so we need to figure that out
